Extensions Manifest V2 support (#3246)
MicrosoftDocs/edge-developer: microsoft-edge/extensions-chromium/developer-guide/manifest-v3.md

* Various minor feature improvements.

<!-- ====================================================================== -->
## Manifest timeline for the Chromium browser engine

Chromium has revised the timelines for ending support for Manifest V2.<!-- todo: add "in/for Chrome"? -->  See:

* [Manifest V2 support timeline](https://developer.chrome.com/docs/extensions/develop/migrate/mv2-deprecation-timeline)
* [The phase-out timeline](https://developer.chrome.com/blog/resuming-the-transition-to-mv3#the_phase-out_timeline) in _Resuming the transition to Manifest V3_.

<!-- ====================================================================== -->
## Manifest timeline for Microsoft Edge and Partner Center

> <!-- todo: condense this h2 section a lot -->

Author: condense this h2 section a lot - move all the takeaway summary points into the table
@sivMSFT

To help you plan the migration of your extension from Manifest V2 to Manifest V3, consider the following schedule for support for Manifest V2 and V3 in Microsoft Edge and Microsoft Partner Center.

> The Microsoft Edge team will independently decide on MV3 migration timelines for Microsoft Edge Add-ons and share an update here.<!-- todo: revise wording in this h2 section? -->  We continue to analyze the concerns raised by extension developers and explore the optimal path for the Microsoft Edge Add-ons ecosystem.<!-- todo: change to "for Microsoft Edge Add-ons."  'ecosystem' is undefined in these articles; if it's a needed term, define it -->  Meanwhile, to plan the migration of your extension from Manifest V2 to Manifest V3, see [Manifest timeline for the Chromium browser engine](#manifest-timeline-for-the-chromium-browser-engine), above.

Author: ```suggestion
The Microsoft Edge team will independently decide on MV3 migration timelines for Microsoft Edge Add-ons and share an update here.<!-- todo: revise wording in this h2 section? -->  We continue to analyze the concerns raised by extension developers and explore the optimal path for Microsoft Edge Add-ons.  Meanwhile, to plan the migration of your extension from Manifest V2 to Manifest V3, see [Manifest timeline for the Chromium browser engine](#manifest-timeline-for-the-chromium-browser-engine), above.
```
'ecosystem' is undefined in these articles; if it's a needed term, define it 

@sivMSFT - techreview this particular change of a couple words

> The Microsoft Edge team will independently decide on MV3 migration timelines for Microsoft Edge Add-ons and share an update here.<!-- todo: revise wording in this h2 section? -->  We continue to analyze the concerns raised by extension developers and explore the optimal path for the Microsoft Edge Add-ons ecosystem.<!-- todo: change to "for Microsoft Edge Add-ons."  'ecosystem' is undefined in these articles; if it's a needed term, define it -->  Meanwhile, to plan the migration of your extension from Manifest V2 to Manifest V3, see [Manifest timeline for the Chromium browser engine](#manifest-timeline-for-the-chromium-browser-engine), above.

Author: "team will ... share an update here"
below, it says instead:
"we will communicate with extension developers when we plan the deprecation of existing MV2-based extensions."

need to bring together & reconcile these two different, similar statements: we will update this page AND communicate [how?] with [all??] extension developers.
@sivMSFT

| TBD | Microsoft Partner Center will no longer accept updates to existing Manifest V2 extensions. Developers can submit updates for migrating a V2 extension to V3. | Microsoft Edge stops running Manifest V2 extensions. Enterprises can allow Manifest V2 extensions to run on Microsoft Edge, by using enterprise policies. |
| TBD | No change. | Manifest V2 extensions will no longer function in Microsoft Edge, even with the use of enterprise policies. |

<!-- ------------------------------ -->

> #### Timeline for MV2-based extensions

Author: ```suggestion
#### Timeline for MV2-based extensions for Enterprise partners
```
not sure how specific/focused the section is, @sivMSFT

> The move to Manifest V3 improves the security, privacy, and performance of extensions.  Our current strategy for Microsoft Edge is to support new MV3-based extensions and support updates to existing MV2-based extensions from our partner ecosystem.  Microsoft Edge Extensions will continue to support updates for MV2-based extensions for an extended period of time, and we will communicate with our partner ecosystem when we plan the deprecation of existing MV2-based extensions.  For enterprises, Manifest V2 extensions will continue to be supported through Enterprise policies at least until the date in the Chromium Manifest V2 support timeline.

Author: the initial draft text: 
MV2 timeline for Enterprise: 
We see the move to MV3 as a change in the positive direction to improve security, privacy and performance of extensions. Our current strategy is to honour any new mv3 extensions from our partner ecosystem and any updates to existing mv2 extensions. At the same time, we will continue updates for MV2 extensions for an extended period of time and will communicate once we plan deprecation of existing mv2 extensions. For enterprises, Manifest V2 extensions will continue to be supported through Enterprise policies at least until the date in the Chromium Manifest V2 support timeline.

@sivMSFT

<!-- ====================================================================== -->
## Providing feedback

> Microsoft continues to iteratively improve the platform and address the feedback shared by extension developers.  Share your questions, comments, and concerns with the team on Twitter [@MSEdgeDev](https://twitter.com/msedgedev/) or via the [Microsoft Edge Insider forums on TechCommunity](https://techcommunity.microsoft.com/t5/articles/manifest-v3-changes-are-now-available-in-microsoft-edge/m-p/1780254).

Author: these two links are not mentioned in 
https://learn.microsoft.com/microsoft-edge/extensions-chromium/publish/contact-extensions-team 
@sivMSFT

| TBD | Microsoft Partner Center will no longer accept updates to existing Manifest V2 extensions. Developers can submit updates for migrating a V2 extension to V3. | Microsoft Edge stops running Manifest V2 extensions. Enterprises can allow Manifest V2 extensions to run on Microsoft Edge, by using enterprise policies. |
| TBD | No change. | Manifest V2 extensions will no longer function in Microsoft Edge, even with the use of enterprise policies. |

> <!-- todo: existing MV2 extensions already published on the Microsoft Edge Add-ons website can be updated until the end of FY-24. [what Month/Day?]  This timeline is subject to change. -->

Author: ```suggestion
Existing MV2 extensions already published on the Microsoft Edge Add-ons website can be updated until December 31, 2024.  This timeline is subject to change.
```
@sivMSFT - add this statement?  if so, this statement should be moved into the summary table.  is date correct?

> <!-- todo: at TBDs, add "see below"?  or, replace right-hand cell content by the main point that's below? -->

Author: in the summary table, replace "tbd" by some other wording, possibly "see below"
all the substantial points in this section should be moved into the summary table.
@sivMSFT

<!-- ------------------------------ -->
#### Timeline for ending support for Manifest V2-based Microsoft Edge extensions for enterprises

> <!-- todo: orig draft wording, for reference:

Author: i'll delete this provided draft wording.  instead of adding such a paragraph to the h2 section, better to condense the h2 section, fully leveraging the summary table.
@sivMSFT

about MV2 TimeLine for Enterprise:
 
We see the move to MV3 as a change in the positive direction to improve security, privacy and performance of extensions.
Our current strategy is to honour any new mv3 extensions from our partner ecosystem and any updates to existing mv2 extensions.
At the same time, we will continue updates for MV2 extensions for an extended period of time and will communicate once we plan deprecation of existing mv2 extensions.
For enterprises, Manifest V2 extensions will continue to be supported through Enterprise policies at least until the date in the Chromium Manifest V2 support timeline.
-->

> The move to Manifest V3 improves the security, privacy, and performance of extensions.<!-- todo: delete sentence about MV3, b/c off-topic for this "MV2" section -->

Author: ```suggestion
```
delete sentence about MV3, b/c off-topic for this "MV2" section

> Our current strategy for Microsoft Edge is to support new MV3-based extensions and support updates to existing MV2-based extensions from our partner ecosystem.<!-- todo: delete sentence b/c the mv3 point is off-topic for this "MV2" section, & the mv2 point is redundant w/ next, clearer sentence --><!-- todo: delete "from our partner ecosystem" b/c it's undefined & redundant or superfluous; not needed: can there exist any existing MV2-based extensions that are not from our partner ecosystem? -->

Author: ```suggestion
```
delete sentence, b/c the mv3 point is off-topic for this "MV2" section, & the mv2 point is redundant w/ the next, clearer sentence

> Our current strategy for Microsoft Edge is to support new MV3-based extensions and support updates to existing MV2-based extensions from our partner ecosystem.<!-- todo: delete sentence b/c the mv3 point is off-topic for this "MV2" section, & the mv2 point is redundant w/ next, clearer sentence --><!-- todo: delete "from our partner ecosystem" b/c it's undefined & redundant or superfluous; not needed: can there exist any existing MV2-based extensions that are not from our partner ecosystem? -->

Author: ```suggestion
Our current strategy for Microsoft Edge is to support new MV3-based extensions and support updates to existing MV2-based extensions.
```
if we keep this sentence, delete "from our partner ecosystem", b/c it's undefined & redundant or superfluous; not needed: can there exist any existing MV2-based extensions that are not from our partner ecosystem?

> Microsoft Edge Extensions will continue to support updates for MV2-based extensions for an extended period of time, and we will communicate with extension developers when we plan the deprecation of existing MV2-based extensions.<!-- todo: move this sentence to be first -->

Author: move this key sentence to be first - or better, move it into the summary table and get rid of this new h4 subsection
@sivMSFT

> For enterprises, Manifest V2 extensions will continue to be supported through Enterprise policies at least until the date in the Chromium Manifest V2 support timeline.<!-- todo: revise phrase "at least ... timeline" -->

Author: revise the phrase "at least until the date in the Chromium Manifest V2 support timeline", b/c that's the problem that this PR is intended to fix & clarify.
@sivMSFT 

that's the sentence that appears in live docs
https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/developer-guide/manifest-v3#manifest-timeline-for-microsoft-edge-and-partner-center

> <!-- todo: delete almost all instances of "we", "our", "us", per https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/we -->

Author: delete almost all instances of "we", "our", "us", per https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/we

<!-- ====================================================================== -->
## Providing feedback

> <!-- todo: when to use twitter, techcommunity, or github forum? -->

Author: when to use twitter, techcommunity, or github forum?
@sivMSFT
